gratipay/billing/exchanges.py

```diff
@@ -82,6 +83,7 @@ def create_card_hold(db, participant, amount):
             'options': { 'submit_for_settlement': False },
             'custom_fields': {'participant_id': participant.id}
         })
+        ref = result.transaction.id
```

Comment:
> all the newly inserted records afterwards will have ref field as non-null?

I believe they will/should be null if the transaction fails. We still want to record our attempt at a transaction even if the API call fails.